// [[Reason: TODO File is not auto-fixed]] ////

//// [tests/cases/compiler/declarationEmitCommonJsModuleReferencedType.ts] ////

===================================================================
--- TSC	declarations
+++ DTE	declarations
@@ -1,8 +1,15 @@
 
+
+//// [r/entry.d.ts]
+import { RootProps } from "root";
+export declare const x: invalid;
+export declare const y: RootProps;
+//# sourceMappingURL=entry.d.ts.map
 /// [Errors] ////
 
 r/entry.ts(3,14): error TS2742: The inferred type of 'x' cannot be named without a reference to 'foo/node_modules/nested'. This is likely not portable. A type annotation is necessary.
+r/entry.ts(3,18): error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
 
 
 ==== r/node_modules/foo/node_modules/nested/index.d.ts (0 errors) ====
     export interface NestedProps {}
@@ -20,12 +27,14 @@
 ==== node_modules/root/index.d.ts (0 errors) ====
     export interface RootProps {}
     
     export function bar(): RootProps;
-==== r/entry.ts (1 errors) ====
+==== r/entry.ts (2 errors) ====
     import { foo } from "foo";
     import { RootProps, bar } from "root";
     export const x = foo();
                  ~
 !!! error TS2742: The inferred type of 'x' cannot be named without a reference to 'foo/node_modules/nested'. This is likely not portable. A type annotation is necessary.
+                     ~~~~~
+!!! error TS9007: Declaration emit for this file requires type resolution. An explicit type annotation may unblock declaration emit.
     export const y: RootProps = bar();
     
\ No newline at end of file
